lib/src/rules/no_boolean_literal_compare_rule.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:dart_code_metrics/src/models/code_issue.dart';
+import 'package:dart_code_metrics/src/models/code_issue_severity.dart';
+
+import 'base_rule.dart';
+import 'rule_utils.dart';
+
+// Inspired by TSLint (https://palantir.github.io/tslint/rules/no-boolean-literal-compare/)
+
+class NoBooleanLiteralCompareRule extends BaseRule {
+  static const _failure =
+      'Comparing boolean values to boolean literals is unnecessary, as those expressions will result in booleans too. Just use the boolean values directly or negate them.';
+
+  static const _useItDirectly =
+      'This expression is unnecessarily compared to a boolean. Just use it directly.';
+  static const _negate =
+      'This expression is unnecessarily compared to a boolean. Just negate it.';
+
+  const NoBooleanLiteralCompareRule()
+      : super(
+          id: 'no-boolean-literal-compare',
+          severity: CodeIssueSeverity.style,
+        );
+
+  @override
+  Iterable<CodeIssue> check(CompilationUnit unit, Uri sourceUrl) {
+    final _visitor = _Visitor();
+
+    unit.visitChildren(_visitor);
+
+    final issues = <CodeIssue>[];
+
+    for (final expression in _visitor.expressions) {
+      String correction;
+      bool useDirect;
+
+      if (expression.leftOperand is! BooleanLiteral) {
+        correction = expression.leftOperand.toString();
+        useDirect = (expression.operator.type == TokenType.EQ_EQ &&
+                expression.rightOperand.toString() == 'true') ||
+            (expression.operator.type == TokenType.BANG_EQ &&
+                expression.rightOperand.toString() == 'false');
+      } else {
+        correction = expression.rightOperand.toString();
```

Comment:
Identical to then branch, except for right/left operand. Maybe extract this logic to subfunction and just pass expression inside?